### ISSUE TITLE
React warning with JSX eventContent "flushSync was called from inside a lifecycle method"

### ISSUE THREAD

**Issue by AmruthPillai** [fullcalendar/fullcalendar#7334]
### Reduced Test Case

-

### Do you understand that if a reduced test case is not provided, we will intentionally delay triaging of your ticket?

- [X] I understand

### Which connector are you using (React/Angular/etc)?

React

### Bug Description

When I updated to the latest version `v6.1.7`, started facing this issue.

There's no visual bug, but I keep getting this warning fired many times when the calendar view updates.

```
Warning: flushSync was called from inside a lifecycle method. React cannot flush when React is already rendering. Consider moving this call to a scheduler task or micro task.
```

![Screenshot 2023-05-16 at 2 13 53 PM](https://github.com/fullcalendar/fullcalendar/assets/1134738/11469e1a-690b-485c-9d1e-d2bad51baece)

**Comment by walerun** [comment]
the same for me 🤔

**Comment by cbaconnier** [comment]
Happening to me as well on React.
For me it seems to come from `eventContent`.

```tsx

  const renderEventContent = (eventInfo: EventContentArg) => {
    return (
      <div>{/* ... */}</div>
    )
  }

// ...

<FullCalendar
    eventContent={renderEventContent}
/>
```

**Comment by River-Hendriksen** [comment]
So I thought I would add to this because I am experiencing this issue also in `v6.1.7`

The issue issue seems to be coming from styles injected into the `renderEventContent` that are dynamic, e.g. `rem`. Here is an example that causes the `flushSync` error (please excuse the inline styles): 

```
const renderEventContent = (eventInfo: EventContentArg) => {
        return (
            <div
                style={{
                    display: 'flex',
                    color: 'black',
                    padding: '.25rem',
                    paddingLeft: '12px',
                    fontSize: '14px',
                    wordBreak: 'normal',
                    whiteSpace: 'pre-line',
                }}
            >
                {eventInfo.event.title}
            </div>
        );
    };
```

here you can see that padding is set to `.25rem` and when i go to load the page it crashes with `flushSync`. If i change padding to a static value: 

```
const renderEventContent = (eventInfo: EventContentArg) => {
        return (
            <div
                style={{
                    display: 'flex',
                    color: 'black',
                    padding: '3px',
                    paddingLeft: '12px',
                    fontSize: '14px',
                    wordBreak: 'normal',
                    whiteSpace: 'pre-line',
                }}
            >
                {eventInfo.event.title}
            </div>
        );
    };
```

Then I can load without the error. Why this occurs I have no idea, probably something with the way fullcalendar renders events, but hopefully this is helpful to someone having this issue!

**Comment by Arithmetics** [comment]
same for me

**Comment by acerix** [comment]
Would you be able to post a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/)? Would really appreciate it because the time saved reproducing will be time spent fixing.

The React example project uses `eventContent` like that but I don't see any warning:

https://stackblitz.com/github/fullcalendar/fullcalendar-examples/tree/main/react

**Comment by cbaconnier** [comment]
> Would you be able to post a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/)? Would really appreciate it because the time saved reproducing will be time spent fixing.
> 
> The React example project uses `eventContent` like that but I don't see any warning:
> 
> https://stackblitz.com/github/fullcalendar/fullcalendar-examples/tree/main/react

This is what I see in the console when I run your example and click on `toggle weekends`

![image](https://github.com/fullcalendar/fullcalendar/assets/4738184/6273a874-db94-4d13-95dc-6323c32eb406)

**Comment by acerix** [comment]
Thanks for the info,  I also see the warning after `toggle weekends`.

**Comment by RomOleg** [comment]
Same with "resourceAreaColumns" when you specify cellContent: () => { return (<MyComponent />) }
![image](https://github.com/fullcalendar/fullcalendar/assets/71003372/ff19bec1-86f7-4050-8beb-d6fc7b21b296)

**Comment by arshaw** [comment]
This is now fixed in [v6.1.8](https://github.com/fullcalendar/fullcalendar/releases/tag/v6.1.8)

Updated repro:
https://stackblitz.com/edit/github-bv3xxu?file=package.json
